examples/rn-connection-and-error/frontend/app/hooks/useDemoSyncTriggers.ts

```diff
@@ -0,0 +1,200 @@
+     * until this listener is invoked.
+     */
+    const handleConnectionChange = (
+      newState: ConnectionState,
+      oldState: ConnectionState,
+    ) => {
+      const connecting = newState === ConnectionState.Connecting;
+      const connected = newState === ConnectionState.Connected;
+      const disconnected =
+        oldState === ConnectionState.Connected &&
+        newState === ConnectionState.Disconnected;
+      /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+      const failedReconnecting =
+        oldState === ConnectionState.Connecting &&
+        newState === ConnectionState.Disconnected;
+
+      if (connecting) {
+        logger.info('Connecting...');
+      } else if (connected) {
+        logger.info('Connected.');
+      } else if (disconnected) {
+        logger.info('Disconnected.');
+
+        // At this point, the `newState` is `ConnectionState.Disconnected`. Automatic retries
+        // will start and the state will alternate in the following way for the period where
+        // there is NO network connection:
+        //    (1) oldState: ConnectionState.Disconnected, newState: ConnectionState.Connecting
+        //    (2) oldState: ConnectionState.Connecting, newState: ConnectionState.Disconnected
+        // Calling `App.Sync.Session.reconnect()` is not needed due to automatic retries.
+      } /* failedReconnecting */ else {
+        logger.info('Failed to reconnect.');
+      }
+
+      setIsConnected(connected);
+    };
+    realm.syncSession?.addConnectionNotification(handleConnectionChange);
+
+    return () =>
+      realm.syncSession?.removeConnectionNotification(handleConnectionChange);
+  }, [realm.syncSession]);
+
+  /**
+   * Trigger the sync error listener by trying to create a `Store` that
+   * is outside of the query filter subscribed to. Since we subscribed
+   * to the store with a given ID (see `App.tsx`), attempting to create
+   * one with a different ID will generate a sync error.
+   *
+   * @note
+   * You can also trigger sync errors by modifying the permissions of
+   * fields and/or collections, and then try to perform non-permitted
+   * operations from the client. To read more about permissions, see:
+   * {@link https://www.mongodb.com/docs/atlas/app-services/rules/roles/#define-roles---permissions}
+   */
+  const triggerSyncError = useCallback(() => {
+    realm.write(() => {
+      const NON_SUBSCRIBED_STORE_ID = new BSON.ObjectId();
+      realm.create(Store, {_id: NON_SUBSCRIBED_STORE_ID});
+    });
+  }, [realm]);
+
+  /**
+   * Trigger the client reset listeners by calling a custom Atlas Function
+   * (see `backend/functions/triggerClientReset.js`) that deletes the client
+   * files for the current user.
+   *
+   * @note
+   * This should NOT be used in production.
+   */
+  const triggerClientReset = useCallback(async () => {
+    await currentUser.functions.triggerClientReset();
```

Comment:
Seems the only way to recover from this is to restart the application.  Is that correct?

Reply:
Either restart or reregister as the trigger function deletes the client files. (Although we still gotta look into our `onBefore` and `onAfter` callbacks not being invoked. Will create a ticket. 👍)